plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExBudgeting.hs

```diff
@@ -67,12 +68,36 @@ type CostModel = CostModelBase CostingFun
    Note also that it's important to perform the same adjustments on the R output
    in TestCostModel.hs so that Haskell and R results agree (which is why
    `toCostUnit` is exported).
+   TODO: this also scales memory costs, which we perhaps don't need to do.  Does
+   that cost us anything?
 -}
 
+{- Note [Time units]. What units are times measured in?  The Criterion output
+   produces times in seconds, and these are usually very small, typically of the
+   order of 10^-6.  In models.R, the get.bench.data funtion multiplies
+   everything by 10^6 after reading it in, so it (and the models it outputs)
+   deal with times in microseconds.  So for example the model for "addInteger"
+   may have an intercept of 0.249487779229322 and a slope of
+   1.87065741871939e-3.  This means that it's taking a basic time of 249ns plus
+   another 1.87ns for every word in the input (in fact, for the maximum of the
+   number of words in the input).  This is still very small, so here we're
+   scaling up by another 10^6, to get times in picoseconds.  For the addInteger
```

Comment:
Why don't we do this in the R code, so everything from then on is just in picoseconds?

Reply:
I'm still not quite sure whether we need to do it in picoseconds yet, but I'll look at that soon for the "use integers instead of floats" issue and may change the R output then.  It is quite easy to get the units wrong and add together ns and ps for example, so making things canonical early is probably best. So yes, I don plan to revisit this, but in a later PR.